src/core/commands/todo/CompleteTodoCommand.ts

```diff
@@ -0,0 +1,22 @@
+import { TodoRepository } from "../../../infrastructure/repositories/TodoRepository";
+import { IResult } from "../../results/successful/SuccessfulResult";
+
+export class CompleteTodoCommand {
+  private _todoRepository: TodoRepository = new TodoRepository();
+
+  public async handle(uid: string): Promise<IResult<void>> {
+    // Todo: fix; this returns new instance of a todo.
+    // The todo, in this example, is not actually updated in backend.
+    // So it returns the old todo which is not updated.
+    // Question is, should we just return the reference (in memory) todo here or refetch it from backend?
+    const todo = await this._todoRepository.getById(uid);
```

Comment:
Should we fetch the new todo here? It is already in memory, or at least the view-model contains information such as completed and title. We could create a new instance of 'TodoEntity' -> complete it and check if it's possible. If it is fetch the API and update backend/database, then wait for successful result of backend. And not refetch the todo, refetching might be unnecessary since we already have the todo as a view model.

Reply:
this returns new instance of a todo. The todo, in this example, is not actually updated in backend. So it returns the old todo which is not updated. Question is, should we just return the reference (in memory) todo here or refetch it from backend?